src/engine/sparqlExpressions/NumericBinaryExpressions.cpp

```diff
@@ -68,9 +66,198 @@ inline auto andLambda = [](TernaryBool a, TernaryBool b) {
   }
   return Id::makeUndefined();
 };
-NARY_EXPRESSION(AndExpression, 2,
-                FV<decltype(andLambda), EffectiveBooleanValueGetter>,
-                SET<SetOfIntervals::Intersection>);
+
+namespace constructPrefilterExpr {
+
+// _____________________________________________________________________________
+template <typename BinaryPrefilterExpr>
+std::optional<std::vector<PrefilterExprVariablePair>> mergeChildrenAndLogicImpl(
+    std::optional<std::vector<PrefilterExprVariablePair>>&& leftChildExprs,
```

Comment:
The name is very confusing, as `AND` appears to be a word here and not an expression.
better something like  `mergeChildrenForAndExpressionImpl` (the same for `OR`).